backend/coins/eth/rpcclient/wrapper/wrapper.go

```diff
@@ -0,0 +1,146 @@
+package wrapper
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/accounts"
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth"
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/rpcclient"
+
+	"github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/erc20"
+	ethtypes "github.com/BitBoxSwiss/bitbox-wallet-app/backend/coins/eth/types"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type rpcClientWithTransactionsSource interface {
+	eth.TransactionsSource
+	rpcclient.Interface
+}
+
+type client struct {
+	defaultRpcClient  rpcClientWithTransactionsSource
+	fallbackRpcClient rpcClientWithTransactionsSource
+
+	// TODO do we maybe want to add some fields to keep track
+	// of which one we are using? if we switch to the fallback one
+	// it might make sense to use that for a while instead of wasting
+	// time each time trying the default first and falling back to the
+	// other one later.
```

Comment:
Definitely! Maybe this package could help, but it may also be overkill :shrug: 

https://github.com/BitBoxSwiss/block-client-go/blob/master/failover/failover.go

Reply:
Good shout, I'll take a look at it before merging to master!